File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/GenericCommands_ViewTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "GenericCommands_ViewModelTestEnvironment.h"
#include <string>
#include "CommandEnum.h"
#include "GenericCommands_ViewModel.h"
#include "GenericCommands_ViewModelTestEnvironmentImpl.h"

namespace commands
{
  class GenericCommands_ViewTests : public testing::Test
  {
  public:
    std::shared_ptr<GenericCommands_ViewModel> sut;
    std::shared_ptr<GenericCommands_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    // When Helper Definitions
    virtual void when_MyCommand();
  protected:
    void SetUp() override;
  };
  void GenericCommands_ViewTests::SetUp()
  {
    this->testEnvironment = std::make_shared<GenericCommands_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  /// Scenario: Generic Commands Call
  ///   given:
  ///    when: MyCommand
  ///    then:
  TEST_F(GenericCommands_ViewTests, Generic_Commands_Call_given_when_MyCommand_then_)
  {
    this->BuildSut();
    this->when_MyCommand();
  }
  void GenericCommands_ViewTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void GenericCommands_ViewTests::when_MyCommand()
  {
    this->sut->myCommand(false, { std::string("A"), std::string("B") }, commands::CommandEnum::Value1);
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/GenericCommands_ViewTests.cs
/// <filename>
///     GenericCommands_ViewTests.cs
/// </filename>
using commands;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class GenericCommands_ViewTests
{
    private commands.GenericCommands_ViewModel sut;

    private commands.GenericCommands_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new commands.GenericCommands_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    /*

    Scenario: Generic Commands Call
      given:
       when: MyCommand
       then:
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void Generic_Commands_Call_given_when_MyCommand_then_()
    {
        this.BuildSut();
        this.when_MyCommand();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    // When Helper Definitions
    public virtual void when_MyCommand()
    {
        this.sut.myCommand("Foo", false, new System.Collections.Generic.List<string> { "A", "B" }, commands.CommandEnum.Value1);
    }
}
